osu.Server.Spectator/Database/DatabaseAccess.cs

```diff
@@ -168,20 +168,19 @@ public async Task RemoveRoomParticipantAsync(MultiplayerRoom room, MultiplayerRo
             }
         }
 
-        public async Task<multiplayer_playlist_item?> GetPlaylistItemFromRoomAsync(long roomId, long playlistItemId)
+        public async Task<multiplayer_playlist_item> GetPlaylistItemAsync(long playlistItemId)
         {
-            return await connection.QueryFirstOrDefaultAsync<multiplayer_playlist_item>("SELECT * FROM multiplayer_playlist_items WHERE id = @Id AND room_id = @RoomId", new
+            return await connection.QueryFirstOrDefaultAsync<multiplayer_playlist_item>("SELECT * FROM multiplayer_playlist_items WHERE id = @Id", new
```

Comment:
Hmm, a few small things here:

* The nullable type annotation was dropped, while the linq query still uses `FirstOrDefaultAsync`. Would prefer either bringing the annotation back, or using `QueryFirst`/`QuerySingle`.
* Also not super sure about dropping the room check. May be better/nicer to keep it for safety, to avoid accidentally performing cross-room fetches. This is also relevant to `MarkPlaylistItemAsPlayedAsync`, which also uses a single playlist item ID rather than playlist item ID + room ID.

Reply:
Fixed the `FirstOrDefaultAsync()`. I intended to change it but forgot.

I personally find providing a room ID superfluous, but I've also applied that change.